libs/template/class/src/lib/class.directive.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  IterableChanges,
+  IterableDiffer,
+  IterableDiffers,
+  KeyValueChanges,
+  KeyValueDiffer,
+  KeyValueDiffers,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+  ɵisListLikeIterable as isListLikeIterable,
+} from '@angular/core';
+import {
+  accumulateObservables,
+  RxStrategyNames,
+  RxStrategyProvider,
+} from '@rx-angular/cdk';
+import { coerceAllFactory } from '@rx-angular/cdk/coercing';
+import {
+  BehaviorSubject,
+  isObservable,
+  Observable,
+  ObservableInput,
+  ReplaySubject,
+  Subscription,
+} from 'rxjs';
+import { map, switchAll, switchMap, withLatestFrom } from 'rxjs/operators';
+
+type NgClassValues =
+  | string[]
+  | Set<string>
+  | Record<string, boolean>
+  | string
+  | null
+  | undefined;
+
+type RxClassInput =
+  | Observable<NgClassValues>
+  | Record<string, Observable<boolean>>;
+
+@Directive({ selector: '[rxClass]' })
+export class ClassDirective implements OnInit, OnDestroy {
+  private iterableDiffer: IterableDiffer<string> | null = null;
+  private keyValueDiffer: KeyValueDiffer<string, any> | null = null;
+  private initialClasses: string[] = [];
+  private rawClass: NgClassValues = null;
+  private newInput = false;
+  private sub = new Subscription();
+
+  constructor(
+    private readonly iterableDiffers: IterableDiffers,
+    private readonly keyValueDiffers: KeyValueDiffers,
+    private readonly ngEl: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly strategyProvider: RxStrategyProvider
+  ) {}
+
+  private rxClass = coerceAllFactory<NgClassValues>(
+    () => new ReplaySubject(1),
+    switchAll()
+  );
+
+  @Input('rxClass')
+  set rxClass$(value$: RxClassInput) {
+    this.newInput = true;
+    this.rxClass.next(
+      (isObservable(value$) ? value$ : accumulateObservables(value$)).pipe(
+        map((value) => (typeof value === 'string' ? [value] : value))
+      )
+    );
+  }
+
+  private strategy = coerceAllFactory(
+    () => new BehaviorSubject<ObservableInput<string> | string>('immediate'),
+    switchAll()
+  );
+
+  @Input('rxClassStrategy')
+  set strategy$(
+    strategy: RxStrategyNames<string> | Observable<RxStrategyNames<string>>
+  ) {
+    this.strategy.next(strategy);
+  }
+
+  ngOnInit(): void {
+    this.sub.add(
+      this.rxClass.values$
+        .pipe(
+          withLatestFrom(this.strategy.values$),
+          switchMap(([value, strategy]) =>
+            this.strategyProvider.schedule(
+              () => {
+                if (this.newInput) {
+                  this.removeClasses(this.rawClass);
+                  this.applyClasses(this.initialClasses);
```

Comment:
Why do we need to run apply here? The Initial Classes are empty so the functions will noop.
Is this maybe related to the classes already existing in the DOM as static strings?

Reply:
I double-checked that and it looks like we don't really need to do this cleanup (most probably just legacy code I forgot to remove).